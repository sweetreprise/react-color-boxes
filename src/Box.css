.box, .container {
    display: inline-block;
}

.container {
    margin: 0.5em;
}

.color-id {
    color: #fff;
    background-color: #000;
    display: table;
    margin: 0 auto;
    font-family: monospace;
}