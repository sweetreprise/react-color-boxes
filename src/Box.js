import React from 'react';
import './Box.css'

const Box = ({id, width, height, bgColor}) => {

    const handleClick = (e) => {
        e.preventDefault();
        e.target.parentElement.remove()
    }

    return (
        <div className='container'>
            <div className='box' data-testid='box' key={id} style={{
            backgroundColor: bgColor,
            width: width,
            height: height
            }}>
                <button data-testid='delete-button' onClick={handleClick}>X</button>
                <span className='color-id'>{bgColor}</span>
            </div>
        </div>
    )
}

export default Box;