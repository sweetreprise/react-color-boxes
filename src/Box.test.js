import React from 'react';
import { render, fireEvent, queryByTestId } from "@testing-library/react";
import Box from './Box';

it("renders component without crashing", () => {
    render(<Box />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});

it("successfully removes a box when the X is clicked", () => {
    const { queryByTestId } = render(<Box />);
    const box = queryByTestId('box');
    const btn = queryByTestId('delete-button')

    // expect box to be present in the document
    expect(box).toBeInTheDocument()
    // fire click event
    fireEvent.click(btn);

    // expect box to no longer be in the document
    expect(box).not.toBeInTheDocument()
})


