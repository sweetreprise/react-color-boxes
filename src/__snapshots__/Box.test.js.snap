// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="box"
      data-testid="box"
    >
      <button
        data-testid="delete-button"
      >
        X
      </button>
      <span
        class="color-id"
      />
    </div>
  </div>
</DocumentFragment>
`;
